compaction/compaction_manager.cc

```diff
@@ -1155,9 +1156,27 @@ namespace compaction {
 class regular_compaction_task_executor : public compaction_task_executor, public regular_compaction_task_impl {
 public:
     regular_compaction_task_executor(compaction_manager& mgr, throw_if_stopping do_throw_if_stopping, table_state& t)
-        : compaction_task_executor(mgr, do_throw_if_stopping, &t, sstables::compaction_type::Compaction, "Compaction")
+        : compaction_task_executor(mgr, do_throw_if_stopping, &t, sstables::compaction_type::Compaction, "Compaction", true)
         , regular_compaction_task_impl(mgr._task_manager_module, tasks::task_id::create_random_id(), mgr._task_manager_module->new_sequence_number(), t.schema()->ks_name(), t.schema()->cf_name(), "", tasks::task_id::create_null_id())
-    {}
```

Comment:
Hmm I don't think there's a concept of iterations (as being proposed here) to regular compaction. Regular compaction is a continuous (eternal) work to reshape the data to satisfy the strategy. It can happen a compaction happens once every 10 minutes, or many compaction rounds can happen one after the other (successive) if the table got compaction starved for some reason.

Reply:
I did it like this because of https://github.com/scylladb/scylladb/pull/14891#issuecomment-1658133066 - we count each iteration in `compaction_manager::stats::completed_tasks`.

Instead of this I could only increment `progress.completed`, not caring about `progress.total` so that it contains the number of processed bytes. I have no other idea how to count the progress

Reply:
@raphaelsc ping.